src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -603,6 +619,35 @@ internal override CSharpAttributeData EarlyDecodeWellKnownAttribute(ref EarlyDec
                 {
                     arguments.GetOrCreateData<ParameterEarlyWellKnownAttributeData>().HasCallerMemberNameAttribute = true;
                 }
+                else if (CSharpAttributeData.IsTargetEarlyAttribute(arguments.AttributeType, arguments.AttributeSyntax, AttributeDescription.CallerArgumentExpressionAttribute))
+                {
+                    // PROTOTYPE(caller-expr): From 333fred review:
+                    // As the language exists today we can do this as an early attribute, but if we do implement the
+                    // nameof proposal (allowing nameof(parameter) in parameters/attributes on methods) I don't believe it
+                    // will end well (this will cause a loop or fail to bind entirely).
+                    // Consider moving binding of the actual attribute data into regular attribute binding.
+                    arguments.GetOrCreateData<ParameterEarlyWellKnownAttributeData>().HasCallerArgumentExpressionAttribute = true;
```

Comment:
>arguments.GetOrCreateData<ParameterEarlyWellKnownAttributeData>() [](start = 20, length = 65)

Cache and reuse?

Reply:
@AlekseyTs Doesn't `GetOrCreateData` already caches the value?

https://github.com/dotnet/roslyn/blob/f6ba23211d9272426214fe055ab8f1e9983e6dd4/src/Compilers/Core/Portable/Symbols/Attributes/DecodeWellKnownAttributeArguments.cs#L28-L36

Reply:
> Doesn't GetOrCreateData already caches the value?

It does, but you are calling it here twice, I suggest capturing the result in a local instead.

---
In reply to: [600229442](https://github.com/dotnet/roslyn/pull/51952#discussion_r600229442) [](ancestors = 600229442)

Reply:
This is no longer needed as I removed `HasCallerArgumentExpression` and kept only the index property based on your feedback. So I'm calling it only once now.